DatadogCore/Sources/Core/DataStore/CoreDataStore.swift

```diff
@@ -0,0 +1,186 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// An entry stored in Data Store files.
+internal struct CoreDataStoreEntry: Equatable {
```

Comment:
`CoreData` is very misfortunate name for this one 😄 
One thing I considered when working on mine was using `LocalStore`. So this would be `CoreLocalStoreEntry`. But maybe just dropping `Core` from everything would be enough.

Not a hard blocker, but IMO `CoreData` thing may confuse future developers.

Reply:
Changed to `FeatureDataStore` as it better follows the agreed design 👍.